app/src/docs/_examples/pickerInput/CascadeSelectionModes.example.tsx

```diff
@@ -26,10 +26,15 @@ export default function CascadeSelectionModesExample() {
         {
             api: (request, ctx) => {
                 const { search } = request;
-                // if search is specified, it is required to search over all the children,
-                // and since parentId is meaningful value, it is required to exclude it from the filter.
-                const filter = search ? {} : { parentId: ctx?.parentId };
-                return svc.api.demo.locations({ ...request, search, filter });
+                if (search && ctx.parentId) { // >1 level, search
```

Comment:
let's also turn on `flattenSearchResults` for this example, like for all others lazy tree pickers, because it's a recommended option